lib/Conversion/TorchOnnxToTorch/DefaultDomainAtoF.cpp

```diff
@@ -1472,4 +1472,99 @@ void mlir::torch::onnx_c::populateDefaultDomainAtoF(
+              binder.getLoc(), rewriter.getType<Torch::IntType>(), extract);
+          dimList.push_back(dim);
+        }
+        Value dimValueList = rewriter.create<Torch::PrimListConstructOp>(
+            binder.getLoc(),
+            Torch::ListType::get(Torch::IntType::get(binder.op->getContext())),
+            dimList);
+        Value noneVal = rewriter.create<Torch::ConstantNoneOp>(binder.getLoc());
+
+        // Get fill_value if it is present.
+        Value value_const;
+        auto attr = binder.op->getAttr("torch.onnx.value");
+        auto resultDType = resultType.getDtype();
+        if (!attr) {
+          if (!resultDType.isa<FloatType>()) {
+            return rewriter.notifyMatchFailure(
+                binder.op,
+                "`value` attr and result type (FloatType) mismatch.");
+          }
+          value_const = rewriter.create<Torch::ConstantFloatOp>(
+              binder.getLoc(), rewriter.getF64FloatAttr(0.0));
+        } else if (attr.isa<mlir::DenseElementsAttr>()) {
+          auto denseAttr = attr.dyn_cast_or_null<DenseElementsAttr>();
+          auto denseAttrEleType = denseAttr.getElementType();
+          if (resultDType.isa<mlir::FloatType>()) {
+            if (!isa<FloatType>(denseAttrEleType)) {
+              return rewriter.notifyMatchFailure(
+                  binder.op,
+                  "`value` attr and result type (FloatType) mismatch.");
+            }
+            float floatValue = denseAttr.getSplatValue<FloatAttr>()
+                                   .getValue()
+                                   .convertToFloat();
+            value_const = rewriter.create<Torch::ConstantFloatOp>(
+                binder.getLoc(), rewriter.getF64FloatAttr(floatValue));
+          } else if (resultDType.isa<mlir::IntegerType>()) {
+            if (!isa<IntegerType>(denseAttrEleType)) {
+              return rewriter.notifyMatchFailure(
+                  binder.op,
+                  "`value` attr and result type (IntegerType) mismatch.");
+            }
+            int64_t intValue = denseAttr.getSplatValue<IntegerAttr>().getSInt();
+            value_const = rewriter.create<Torch::ConstantIntOp>(
+                binder.getLoc(), rewriter.getI64IntegerAttr(intValue));
+          } else {
+            return rewriter.notifyMatchFailure(
+                binder.op,
+                "Non int / float type `value` attr tensor not supported.");
+          }
+        } else if (attr.isa<mlir::FloatAttr>()) {
+          float value = attr.dyn_cast<FloatAttr>().getValue().convertToFloat();
+          value_const = rewriter.create<Torch::ConstantFloatOp>(
+              binder.getLoc(), rewriter.getF64FloatAttr(value));
+        } else if (attr.isa<mlir::IntegerAttr>()) {
+          int64_t value = attr.dyn_cast<IntegerAttr>().getSInt();
+          value_const = rewriter.create<Torch::ConstantIntOp>(
+              binder.getLoc(), rewriter.getI64IntegerAttr(value));
+        } else {
```

Comment:
Just another observation. By doing the following up front
```
if (!isa<DenseElementsAttr, FloatAttr, IntegerAttr>) {
          return rewriter.notifyMatchFailure(
              binder.op,
              "`value` attr not float / int / tensor. not supported yet.");
}
```

We can validate the inputs / support upfront, clearly indicating what is supported.

Reply:
Updated 🙏